src/app/components/modal-create-project/modal-create-project.component.ts

```diff
@@ -60,43 +59,52 @@ export class ModalCreateProjectComponent implements OnInit, AfterContentChecked
 
   onSystemSelect(system: any) {
     this.selectedSystem = system;
-    if (system.id.startsWith('project')) {
-      this.projCreateForm.get('linkProject').setValue(true);
-    } else {
-      this.projCreateForm.get('linkProject').setValue(false);
-    }
+    this.projCreateForm.get('watchContent').setValue(false);
   }
 
-  createRapidProject() {
-    this.errorMessage = '';
-    const req: RapidProjectRequest = new RapidProjectRequest(this.rapidFolder.system, this.rapidFolder.path);
-    this.projectsService.createRapidProject(req).subscribe( (project) => {
-      this.close(project);
-    }, (err) => {
-     this.errorMessage = err.toString();
-    });
+  setFileName(ev: any) {
+    this.projCreateForm.get('fileName').setValue(ev.target.value);
   }
 
   submit() {
     this.submitting = true;
     const p = new Project();
+    const pr = new ProjectRequest();
+
     p.description = this.projCreateForm.get('description').value;
     p.name = this.projCreateForm.get('name').value;
-    this.projectsService.create(p).subscribe( (project) => {
-      if (this.projCreateForm.get('exportMapLink').value) {
-        const path = this.selectedFiles.length > 0 ? this.selectedFiles[0].path : this.currentPath;
-        const systemId = this.selectedSystem.id;
-        this.projectsService.exportProject(project,
-                                           systemId,
-                                           path,
-                                           this.projCreateForm.get('linkProject').value,
-                                           this.projCreateForm.get('fileName').value)
-      }
+    p.system_path = this.selectedFiles.length > 0 ? this.selectedFiles[0].path : this.currentPath;
+    p.system_id = this.selectedSystem.id;
+    p.system_file = this.projCreateForm.get('fileName').value
+      ? this.projCreateForm.get('fileName').value
+      : p.name;
+
+    if (this.selectedSystem.id.includes('project')) {
+      pr.observable = true;
+      pr.watch_content = this.projCreateForm.get('watchContent').value;
+    } else {
+      pr.observable = this.projCreateForm.get('watchContent').value;
+      pr.watch_content = pr.observable;
+    }
+
+    this.errorMessage = '';
+
+    pr.project = p;
+
+    this.projectsService.create(pr).subscribe((project) => {
       this.close(project);
     }, err => {
-      this.errorMessage = err.toString();
+      this.errorMessage = err.error.message;
+      this.submitting = false;
     });
-    this.submitting = true;
+  }
+
+  fileAlreadyExists(fileName: string) {
+    return this.filesList.includes(`${fileName}.hazmapper`);
+  }
```

Comment:
```suggestion
```

should we drop for now?

and when we do bring it back into the code base, we got to be careful as our file list isn't always complete for directories with large number of files  (e.g. as user scrolls, they get more of the file listing) .  i wonder if just doing a list for the single file would be a better approach

Reply:
I'll stash it locally and make a ticket 👍🏼